runtime/src/mem/jemalloc/mem-jemalloc.c

```diff
@@ -28,15 +28,29 @@
 #include "chpl-comm.h"
 #include "chpl-mem.h"
 #include "chplmemtrack.h"
+#include "chplsys.h"
+#include "chpl-topo.h"
 #include "chpltypes.h"
 #include "error.h"
 
-static struct shared_heap {
+
+//
+// Information about shared heaps gotten from the comm layer.
+// MAX_HEAPS must be >= the number of sublocales.
+//
+#define MAX_HEAPS 4
```

Comment:
Hmm, hadn't thought about that option.  Would `useUpMemNotInHeap()` then have to do more work, since we'd initialized jemalloc?  Or is that not really a concern?